gui/main.js

```diff
@@ -0,0 +1,60 @@
+
+let { app } = require('electron')
+
+const HomeBrowserWindow = require('./src/home/home-browser-window')
+const ViewerBrowserWindow = require('./src/viewer/viewer-browser-window')
+const EventBus = require('./src/util/event-bus')
+
+let homeWindow = undefined
+const appConfig = {
+    title: "APK Dependency Graph",
+    devMode: true
+}
+
+const ACTION_OPEN_VIEWER_PAGE = 'open-viewer-page'
+const ACTION_OPEN_HOME_PAGE = 'open-home-page'
+
+EventBus.on(ACTION_OPEN_HOME_PAGE, () => {
+    openHomePage()
+})
+
+EventBus.on(ACTION_OPEN_VIEWER_PAGE, () => {
+    openViewerPage()
+    closeHomePage()
+})
+
+app.on('ready', openHomePage)
+app.on('activate', openHomePage)
+app.on('window-all-closed', closeAllWindows)
+
+function isAvaliable(window) {
+    return window !== undefined && window !== null
+}
+
+function openHomePage() {
+    if (!isAvaliable(homeWindow)) {
+        console.log("Opening home page")
+        homeWindow = new HomeBrowserWindow(app, appConfig)
+        homeWindow.open()
+    }
+}
+
+function closeHomePage() {
+    if (isAvaliable(homeWindow)) {
+        console.log("closing home page")
+        homeWindow.close()
+        homeWindow = undefined
+    }
+}
+
+function openViewerPage() {
+    console.log("Opening viewer page")
+    let viewerPage = new ViewerBrowserWindow(appConfig)
+    viewerPage.open()
+}
+
+function closeAllWindows() {
+    if (process.platform !== 'darwin') {
```

Comment:
what it means?

Reply:
When you click on the 'x' to close the window, this functions is called... The idea is to close all the opened windows (if exists more than one window opened), and only do that for the platforms different from Mac OS ('darwin').

Reply:
On Mac OS, when you close a window of an application... the app is not quitted (removed from memory)... it is maintained in the memory to better user experience! So when all application windows are closed it is necessary to quit the app from memory, except for Mac OS. 

This is a recommendation "On macOS, it is common for applications and their menu bar to stay active until the user quits explicitly with Cmd + Q"

Sorry, I wrong in saying this:
> The idea is to close all the opened windows (if exists more than one window opened), and only do that for the platforms different from Mac OS ('darwin').